File: models/map_renderer.py
from os import path
from bokeh.models.layouts import LayoutDOM
from bokeh.core.properties import Instance, List
from bokeh.util.compiler import FromFile

script_path = path.dirname(path.abspath( __file__ ))


class MapRenderer(LayoutDOM):    
    __implementation__ = FromFile(script_path + '/map_renderer.js')
    # feature_locs = List(Instance(FeatureLocation))

File: models/map_renderer.js
import * as _ from "underscore"
import * as $ from "jquery"
import * as p from 'core/properties';
import {LayoutDOM, LayoutDOMView} from 'models/layouts/layout_dom';


class MapRendererView extends LayoutDOMView {
  
  constructor(options) {
    console.log('MapRendererView');
    super(options);
    this.render();
  }
  
  render() {
    console.log('render()');
    this.$el.html(`MapRenderer: ${this.model.name}`);
  }
}


export class MapRenderer extends LayoutDOM {}
MapRenderer.prototype.default_view = MapRendererView;
MapRenderer.prototype.type = 'MapRenderer';
